src/evergreen/task.py

```diff
@@ -274,6 +289,34 @@ def get_tests(
             execution=self.execution if execution is None else execution,
         )
 
+    def get_execution_tasks(
+        self, filter_fn: Optional[Callable[["Task"], bool]] = None
+    ) -> List["Task"]:
+        """
+        Get a list of execution tasks associated with this task.
+
+        If this is a display task, return the tasks execution tasks associated with it. 
+        If this is not a display task, return an empty list.
+
+        :param filter_fn: Function to filter returned results.
+        :return: List of execution tasks.
+        """
+        if self.display_only:
+            execution_tasks = [
+                self._api.task_by_id(task_id, fetch_all_executions=True)
+                for task_id in self.execution_tasks
+            ]
+
+            execution_tasks = [
+                task.get_execution_or_self(self.execution) for task in execution_tasks
+            ]
+
+            if filter_fn:
+                return [task for task in execution_tasks if filter_fn(task)]
+            return execution_tasks
+
+        return []
```

Comment:
Will a display task always have at least one execution task associated with it?

Even though this behavior is documented, returning an empty list seems like it might be a little surprising if the answer to the above question is no. What would you think about returning `None` or raising an exception if you try to get execution tasks from a non display task? That way the different cases are clearly delineated.

Reply:
I've updated to return `None` if we get a non-display task.